hnn_core/hnn_io.py

```diff
@@ -328,25 +329,33 @@ def write_network_configuration(net, fname, overwrite=True):
 
     Parameters
     ----------
-    %(net)s
-    %(fname)s
-    %(overwrite)s
+    net : Network
+        hnn-core Network object
+    output : str, Path, or StringIO
+        Path or buffer to write outputs
+    overwrite : bool
+        Overwrite file if it exists. Default: True
 
-    Yields
-    ------
-    A json file containing the Network configurations.
+    Returns
+    -------
+    None
     """
 
-    if overwrite is False and os.path.exists(fname):
-        raise FileExistsError('File already exists at path %s. Rename '
-                              'the file or set overwrite=True.' % (fname,))
-
     net_data = net.to_dict(write_output=False)
     net_data_converted = _convert_np_array_to_list(net_data)
 
-    # Saving file
-    with open(fname, 'w', encoding='utf-8') as f:
-        json.dump(net_data_converted, f, ensure_ascii=False, indent=4)
+    if isinstance(output, (str, Path)):
+        if overwrite is False and os.path.exists(output):
+            raise FileExistsError('File already exists at path %s. Rename '
```

Comment:
Unit test for this warning?

Reply:
Yep! There's a test for this in test_io.